data/migrations/20190321220524_createIngredientsTable.js

```diff
@@ -0,0 +1,22 @@
+
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable('ingredients', tbl => {
+    tbl.increments();
+    tbl.string('ingredient')
+       .notNullable();
+
+    tbl.float('qty')
+       .notNullable();
+
+     tbl.integer('recipe_id')
+        .unsigned()
+        .references('id')
+        .inTable('recipes')
+        .onDelete('CASCADE')
+        .onUpdate('CASCADE');
```

Comment:
You use this same sort of chain in this migration and also the one for recipes. Why not use a mutual helper function for it? For example:
```
data:
  helpers:
    foreignId
  migrations:
    ...
    recipes
    ingredients
  ...
```

data/helpers/foreignId:
```javascript
const foreignId = (tbl, tblName, single) => {
  tbl.integer(`${single}_id`)
    .unsigned()
    .references('id')
    .on(tblName)
    .onDelete('CASCADE')
    .onUpdate('CASCADE');
};

module.exports = foreignId;
```

data/migrations/recipes:
```javascript
const foreignId = require('../helpers/foreignId');

// ...
    tbl.string('recipe')
      .notNullable()
      .unique();
  
    foreignId(tbl, 'recipes', 'recipe');
  })
// ...
```

and I assume you can figure out the usage in your ingredients migration from there.